opentracing-web-servlet-filter/src/main/java/io/opentracing/contrib/web/servlet/filter/TracingFilter.java

```
@@ -0,0 +1,159 @@
+    /**
+     * Use as a key of ${@link ServletContext#setAttribute(String, Object)} to set span decorators
+     */
+    public static final String SPAN_DECORATORS = TracingFilter.class.getName() + ".spanDecorators";
+    /**
+     * Use as a key of ${@link ServletContext#setAttribute(String, Object)} to set tracing decision
+     */
+    public static final String TRACING_DECISION = TracingFilter.class.getName() + ".tracingDecision";
+    /**
+     * Used as a key of ${@link HttpServletRequest#setAttribute(String, Object)} to inject current server span
+     */
+    public static final String CURRENT_SPAN = TracingFilter.class.getName() + ".currentSpan";
+
+    private FilterConfig filterConfig;
+
+    private Tracer tracer;
+    private List<SpanDecorator> spanDecorators;
+    private TracingDecision tracingDecision;
+
+    /**
+     * When using this constructor one has to provide required (${@link TracingFilter#TRACER},
+     * optional ${@link TracingFilter#SPAN_DECORATORS}, optional ${@link TracingFilter#TRACING_DECISION})
+     * parameters in ${@link ServletContext#setAttribute(String, Object)}.
+     */
+    public TracingFilter() {}
+
+    /**
+     *
+     * @param tracer tracer
+     * @param spanDecorators decorators
+     * @param tracingDecision tracing decision
+     */
+    public TracingFilter(Tracer tracer, List<SpanDecorator> spanDecorators, TracingDecision tracingDecision) {
+        this.tracer = tracer;
+        this.spanDecorators = new ArrayList<>(spanDecorators);
+        this.tracingDecision = tracingDecision;
+    }
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+        this.filterConfig = filterConfig;
+
+        // init if the tracer is null
+        if (null == tracer) {
+            ServletContext servletContext = filterConfig.getServletContext();
+            this.tracer = (Tracer)servletContext.getAttribute(TRACER);
```

Isn't it better to throw an exception. Someone wanted to use the fiter but did not initialized it properly.